test/integration/suites/upgrade/02-verify-codebase-version-is-updated

```diff
@@ -1,40 +1,60 @@
 #!/bin/bash
 
-check-version-against-prior-tag() {
-    # Get nearest prior tag in this branch
-    _nearest_prior_tag=$(git --git-dir $REPODIR/.git describe --match "v*" --abbrev=0 | cut -c 2-)
+check-version-against-latest-release() {
+    # This should be set by the CI/CD system based on the target branch of
+    # the PR.
+    if [ -n "${TEST_TARGET_BRANCH}" ]; then
+        log-info "target branch (explicit): ${TEST_TARGET_BRANCH}"
+    else
+        TEST_TARGET_BRANCH="$(git --git-dir "${REPODIR}"/.git branch --show-current)"
```

Comment:
Hmm, if I understood this correctly, it seems like if you want to run this test in a local branch (i.e. during development), you have to set `TEST_TARGET_BRANCH=master` in your environment in order for the test to pick the right version tag. Should we set `TEST_TARGET_BRANCH` to the default branch (`master`) here instead so that setting that environment variable is not required locally?

Reply:
The intent was that if you were running this in a release branch (i.e. v0.11) that it would do the version check the latest release from that branch. It does make it weird if you run it from a local development branch.

Maybe instead we try first to detect the tracking branch and then fall back to this logic?

That way it would:
branch_to_look_at=tracking branch if set, or if not, current branch
If branch_to_look_at = default branch - check against latest project release
If branch_to_look_at = version branch - check against the latest tag on branch_to_look_at